pkg/helios/cas/cas.go

```diff
@@ -0,0 +1,398 @@
+// Copyright 2025 Oppie Thunder Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// SPDX-License-Identifier: Apache-2.0
+
+package cas
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"github.com/good-night-oppie/helios-engine/pkg/helios/types"
+	"lukechampine.com/blake3"
+)
+
+// writeOp represents a background write operation
+type writeOp struct {
+	filePath string
+	content  []byte
+}
+
+// ContentAddressableStore defines the interface for content-addressable storage
+// implementing the core CAS operations with performance guarantees from research
+type ContentAddressableStore interface {
+	// Store saves content and returns its content-addressable hash
+	// Performance target: <1ms for typical code file sizes
+	Store(content []byte) (types.Hash, error)
+
+	// Load retrieves content by its hash
+	// Performance target: <5ms for typical code file sizes  
+	Load(hash types.Hash) ([]byte, error)
+
+	// Exists checks if content with given hash exists
+	// Performance target: <50μs for existence checks (relaxed during implementation)
+	Exists(hash types.Hash) bool
+
+	// Close releases resources
```

Comment:
API design: consider context-aware methods (e.g., Store(ctx, ...), Load(ctx, ...)) to support cancellation/timeouts during I/O and clean shutdown. It also makes shutdown coordination simpler than global flags.